### PR TITLE
use SelectedInCaptureFunctions in Live tab and when saving the capture

### PR DESCRIPTION
**Pull request by IrinaShkviro**
Fix for http://b/163014195

Introduce `Capture::GSelectedInCaptureFunctions` to store the functions that were hooked in the capture, and show these functions on the "live" tab and when saving capture.

If we want to refer to currently hooked functions we should use `GSelectedFunctionsMap`.

### CODE REVIEW

File: OrbitGl/LiveFunctionsDataView.h [hunk 1]
```diff
@@ -34,7 +34,7 @@ class LiveFunctionsDataView : public DataView {
   std::pair<TextBox*, TextBox*> GetMinMax(
       const orbit_client_protos::FunctionInfo& function) const;
 
-  std::vector<orbit_client_protos::FunctionInfo*> m_Functions;
+  std::vector<std::shared_ptr<orbit_client_protos::FunctionInfo>> m_Functions;
```
Comment (ronaldfw) [inline OrbitGl/LiveFunctionsDataView.h:37]: Nit: Maybe take the opportunity and change m_Functions to functions_?

File: OrbitCaptureClient/CaptureClient.cpp [hunk 2]
```diff
@@ -15,8 +15,7 @@ ABSL_DECLARE_FLAG(bool, frame_pointer_unwinding);
 using orbit_client_protos::FunctionInfo;
 
 void CaptureClient::Capture(
-    int32_t pid,
-    const std::vector<std::shared_ptr<FunctionInfo>>& selected_functions) {
+    int32_t pid, const std::map<uint64_t, FunctionInfo*>& selected_functions) {
```
Comment (florian-kuebler) [inline OrbitCaptureClient/CaptureClient.cpp:18]: Did this actually led to instrumenting a function that was unselected?

File: OrbitGl/App.cpp [hunk 3]
```diff
@@ -584,8 +584,8 @@ bool OrbitApp::StartCapture() {
   }
 
   int32_t pid = Capture::GProcessId;
-  std::vector<std::shared_ptr<FunctionInfo>> selected_functions =
-      Capture::GSelectedFunctions;
+  std::map<uint64_t, FunctionInfo*> selected_functions =
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply OrbitGl/App.cpp:587]: I guess we should postpone this. In capture there are a lot of std::maps where we should think of using the absl versions. Changing them would require a lot of places to be modified. Further we want to change them to get rid of the global anyways. So I suggest to do this together.

File: OrbitGl/LiveFunctionsDataView.h (hunk 1 above)
Comment (IrinaShkviro) [reply OrbitGl/LiveFunctionsDataView.h]: Done, thanks!

File: OrbitGl/LiveFunctionsDataView.cpp [hunk 4]
```diff
@@ -302,17 +303,12 @@ void LiveFunctionsDataView::DoFilter() {
...
-
   m_Functions.clear();
-  for (auto& pair : Capture::GVisibleFunctionsMap) {
-    const ULONG64& address = pair.first;
-    FunctionInfo* func = Capture::GVisibleFunctionsMap[address];
-    m_Functions.push_back(func);
+  size_t functions_count = Capture::GSelectedInCaptureFunctions.size();
+  m_Indices.resize(functions_count);
+  for (size_t i = 0; i < functions_count; ++i) {
+    m_Indices[i] = i;
```
Comment (florian-kuebler) [inline OrbitGl/LiveFunctionsDataView.cpp:310]: If you go for renaming [m_Functions](https://github.com/google/orbit/pull/916/files#r466394460), maybe also go for m_Indices to indices_.

File: OrbitGl/App.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (IrinaShkviro) [reply OrbitGl/App.cpp:587]: I think so... We should also change the type of Capture::GSelectedFunctionsMap. But probably, I would prefer to leave it as is for now as it's not necessary for this fix

[the remark this answers is not in the document]
Comment (ronaldfw) [reply OrbitGl/App.cpp:587]: Ok, makes sense!

File: OrbitGl/LiveFunctionsDataView.cpp (hunk 4 above)
Comment (IrinaShkviro) [reply OrbitGl/LiveFunctionsDataView.cpp]: Done, thanks!

**Pull request merged**